src/Microsoft.Health.Fhir.Core/Features/Operations/FhirOperationDataStoreBase.cs

```diff
@@ -0,0 +1,254 @@
+                            {
+                                record.FailureDetails = new JobFailureDetails(record.FailureDetails.FailureReason + "\r\n" + processResult.FailureDetails.FailureReason, record.FailureDetails.FailureStatusCode);
+                            }
+                        }
+                    }
+                    else
+                    {
+                        if (record.FailureDetails == null)
+                        {
+                            record.FailureDetails = new JobFailureDetails("Processing job had no results", HttpStatusCode.InternalServerError);
+                        }
+                        else
+                        {
+                            record.FailureDetails = new JobFailureDetails(record.FailureDetails.FailureReason + "\r\nProcessing job had no results", HttpStatusCode.InternalServerError);
+                        }
+                    }
+                }
+
+                record.Status = OperationStatus.Failed;
+                status = JobStatus.Failed;
+                result = JsonConvert.SerializeObject(record);
+            }
+            else if (!inFlightJobsExist) // no failures here
+            {
+                foreach (var job in groupJobs.Where(_ => _.Id != jobInfo.Id))
+                {
+                    if (!string.IsNullOrEmpty(job.Result) && !job.Result.Equals("null", StringComparison.OrdinalIgnoreCase))
+                    {
+                        var processResult = JsonConvert.DeserializeObject<ExportJobRecord>(job.Result);
+                        foreach (var output in processResult.Output)
+                        {
+                            if (record.Output.TryGetValue(output.Key, out var exportFileInfos))
+                            {
+                                exportFileInfos.AddRange(output.Value);
+                            }
+                            else
+                            {
+                                record.Output.Add(output.Key, output.Value);
+                            }
+                        }
+                    }
+                }
+
+                record.Status = OperationStatus.Completed;
+                status = JobStatus.Completed;
+                result = JsonConvert.SerializeObject(record);
+            }
+            else
+            {
+                status = JobStatus.Running;
+            }
+        }
+
+        return CreateExportJobOutcome(jobId, result ?? def, jobInfo.Version, (byte)status, jobInfo.CreateDate);
+    }
+
+    public virtual async Task<ExportJobOutcome> UpdateExportJobAsync(ExportJobRecord jobRecord, WeakETag eTag, CancellationToken cancellationToken)
+    {
+        EnsureArg.IsNotNull(jobRecord, nameof(jobRecord));
+
+        if (jobRecord.Status != OperationStatus.Canceled)
+        {
+            throw new NotSupportedException($"Calls to this method with status={jobRecord.Status} are deprecated.");
```

Comment:
This exception is preventing legacy jobs from completing. The legacy jobs will still call this method to complete or fail, but won't get to the JobNotExistException below that will get caught. This exception also needs to be caught in the CosmosFhirOperationDataStore and call the legacy method.

Reply:
Thanks for finding this, updated the logic for detecting legacy jobs in the Cosmos method